playbooks/ansible-changelog-fragment/run.yaml

```diff
@@ -1,15 +1,10 @@
 ---
 - hosts: localhost
   gather_facts: false
+  vars:
+    check_changelog_fragment__git_directory: "{{ zuul.executor.src_root }}/{{ zuul.project.canonical_name }}"
+    check_changelog_fragment__target_branch: "{{ zuul.project.default-branch }}"
   tasks:
-    - name: Check for changelog fragments
-      args:
-        chdir: "{{ zuul.executor.src_root }}/{{ zuul.project.canonical_name }}"
-      shell: git show --name-status --exit-code --oneline --diff-filter=A | grep changelogs/fragments/
```

Comment:
this is a pretty large refactor, I think you could just change this too
```
git show HEAD...origin/{{zuul.branch}} --name-status --exit-code --oneline --diff-filter=A
```
And it will work as expected

Reply:
this does not cover this:
> You do not have to add a changelog fragment for PRs that add new modules and plugins, because our tooling does that for you automatically.

Reply:
That is fine for now, the original purpose of this check was to enforce each commit to specific folders had the fragment. There shouldn't be harm for a user adding it in the same commit as a new module.

Reply:
Also, a commit that only changes tests/ should not require a changelog fragment.

Reply:
> Also, a commit that only changes tests/ should not require a changelog fragment.

That is managed via https://github.com/ansible/project-config/blob/master/zuul.d/jobs.yaml#L76 if a project doesn't want them on tests, I would say create a new job, and parent to ansible-changelog-fragment updating the file matcher.  In the case of network collections, changelogs for tests are still fine.

Reply:
> There shouldn't be harm for a user adding it in the same commit as a new module.

You're right, and in fact there is no harm for a user adding it in the same commit as a new module. But there shouldn't be harm for a user NOT adding it in the same PR/commit as a new module.

Reply:
my feelings here are, that could be an optimization we make in a follow up patch. Even have a discussion on how we want to enhance this job. Your original issue was 'don't only check the last commit', which I think we can land now and test.